analytic_partner/models/account_analytic_line.py

```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, fields
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    partner_id = fields.Many2one(
+        comodel_name='res.partner', string="Partner",
+        domain="[('parent_id', '=', False)]")
```

Comment:
If the rationale is to show only commercial partners, then you may want to filter on either top level partner ór 'is_company' == True.